tests/unit_test/gmail_data/get_gmail_data_test.py

```diff
@@ -0,0 +1,219 @@
+from unittest.mock import MagicMock
+import pytest
+
+from data_pipeline.gmail_data.get_gmail_data import (
+    get_dataframe_for_label_response,
+    get_dataframe_for_thread_response,
+    get_dataframe_for_history_response,
+    iter_link_message_thread_ids,
+    iter_gmail_history,
+    get_label_list,
+    get_one_thread
+)
+
+USER_ID1 = 'user_id1'
+IMPORTED_TIMESTAMP = '2021-04-16'
+
+THREAD_ID1 = 'thread_id1'
+
+START_ID1 = 'start_id1'
+
+
+@pytest.fixture(name='gmail_service_mock')
+def _gmail_service_mock() -> MagicMock:
+    return MagicMock(name='service')
+
+
+@pytest.fixture(name='gmail_label_list_mock')
+def _gmail_label_list_mock(gmail_service_mock: MagicMock) -> MagicMock:
+    users_mock = gmail_service_mock.users
+    labels_mock = users_mock.return_value.labels
+    list_mock = labels_mock.return_value.list
+    return list_mock
+
+
+@pytest.fixture(name='gmail_message_ids_list_mock')
+def _gmail_message_ids_list_mock(gmail_service_mock: MagicMock) -> MagicMock:
+    users_mock = gmail_service_mock.users
+    messages_mock = users_mock.return_value.messages
+    list_mock = messages_mock.return_value.list
+    return list_mock
+
+
+@pytest.fixture(name='gmail_thread_get_mock')
+def _gmail_thread_get_mock(gmail_service_mock: MagicMock) -> MagicMock:
+    users_mock = gmail_service_mock.users
+    threads_mock = users_mock.return_value.threads
+    get_mock = threads_mock.return_value.get
+    return get_mock
+
+
+@pytest.fixture(name='gmail_history_list_mock')
+def _gmail_history_list_mock(gmail_service_mock: MagicMock) -> MagicMock:
+    users_mock = gmail_service_mock.users
+    history_mock = users_mock.return_value.history
+    list_mock = history_mock.return_value.list
+    return list_mock
+
+
+class TestGetLabelList:
+
+    def test_should_pass_user_id_to_list_method(
+            self,
+            gmail_service_mock,
+            gmail_label_list_mock):
+
+        get_label_list(gmail_service_mock, USER_ID1)
+        gmail_label_list_mock.assert_called_with(userId=USER_ID1)
+
+
+# pylint: disable=invalid-name
+class TestGetDataframeForLabelResponse:
+
+    def test_should_pass_user_id_and_imported_timestamp_in_method(self):
+        label_response = {"labels": [{"id": "id1", "name": "name1", "type": "type1"}]}
+        df = get_dataframe_for_label_response(label_response, USER_ID1, IMPORTED_TIMESTAMP)
+        assert df['user_id'].values == [USER_ID1]
+        assert df['imported_timestamp'].values == [IMPORTED_TIMESTAMP]
+
+    def test_should_return_elements_in_label_response(self):
```

Comment:
minor: this is also more subjective. I wasn't sure what *elements* means here. I would maybe name it something like `test_should_extract_selected_fields_from_label_response`